azurelinuxagent/common/logcollector.py

```diff
@@ -274,28 +313,27 @@ def collect_logs(self):
         files_to_collect = []
 
         try:
-            # Clear previous run's output and create base directories if they don't exist already
-            rm_files(_OUTPUT_RESULTS_FILE_PATH)
-            mkdir(_TRUNCATED_FILES_DIR)
-            mkdir(_LOG_COLLECTOR_DIR)
+            # Clear previous run's output and create base directories if they don't exist already.
+            self._create_base_dirs()
+            write_file(_OUTPUT_RESULTS_FILE_PATH, "")
```

Comment:
Wont the logger take care of creating this?

Reply:
The logger wouldn't clear the file, just keep appending to it. I am keeping the `results.txt` file once per run.

Reply:
Ahh ok. 
A very small NIT: Add a wrapper function over this called `reset_file` or something to make it a bit more clearer. Totally fine if you dont do it though, just looks more cleaner to me :)

Reply:
Renamed for clarity.